mergin/client.py

```diff
@@ -482,5 +468,43 @@ def backup_if_conflict(path, checksum):
             move_file(local_path(file["path"]), local_path(file["new_path"]))
 
         local_info["files"] = server_info["files"]
-        local_info["version"] = server_info["version"]
+        local_info["version"] = server_info["version"] if len(server_info["version"]) else 'v0'
         save_project_file(directory, local_info)
+
+    def _download_file(self, project_path, project_version, file, directory):
+        """
+        Helper to download single project file from server in chunks.
+
+        :param project_path: Project's full name (<namespace>/<name>)
+        :type project_path: String
+        :param project_version: Version of the project (v<n>)
+        :type project_version: String
+        :param file: File metadata item from Project['files']
+        :type file: dict
+        :param directory: Project's directory
+        :type directory: String
+        """
+        chunk_size = 10 * 1024 * 1024
+        query_params = {
+            "file": file['path'],
+            "version": project_version
+        }
+        file_dir = os.path.dirname(os.path.normpath(os.path.join(directory, file['path'])))
+        basename = os.path.basename(file['path'])
+        length = 0
+        count = 0
+        while length < file['size']:
+            range_header = {"Range": "bytes={}-{}".format(length, length + chunk_size)}
+            resp = self.get("/v1/project/raw/{}".format(project_path), data=query_params, headers=range_header)
+            # TODO some kind of recovery? do_request already raises exception
```

Comment:
Rather to open an issue than have TODO in a production code. It seems like `do_request` really raises a `ClientError`. Shouldn't be caught here and a while loop suppose to continue (try to resent request again and maybe after a few attempt be shut down)?